x/oracle/types/validatorPerNetwork.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// UpdateValidator reset validator's power
+func (list *ValidatorWhiteList) UpdateValidator(validator sdk.ValAddress, power uint32) {
+	list.GetWhiteList()[validator.String()] = power
+}
+
+// GetValidatorPower return validator's power
+func (list *ValidatorWhiteList) GetValidatorPower(validator sdk.ValAddress) uint32 {
+	if list.ContainValidator(validator) {
+		return list.GetWhiteList()[validator.String()]
+	}
+
+	return 0
+}
+
+// ContainValidator return if validator in the map
+func (list *ValidatorWhiteList) ContainValidator(validator sdk.ValAddress) bool {
+	_, ok := list.GetWhiteList()[validator.String()]
+	return ok
+}
+
+// GetPowerRatio return the power ratio of input validator address list
+func (list *ValidatorWhiteList) GetPowerRatio(claimValidators map[string][]sdk.ValAddress) (string, float64, float64) {
```

Comment:
We should remove this code and just have a helper function like this that that just returns the ratio of power signed off on this particular prophecy.

Reply:
We don't need to make any assumptions about competing validators voting for different prophecies which is where most of this code is.